lib/src/realm_object.dart

```diff
@@ -123,19 +123,43 @@ class RealmCoreAccessor implements RealmAccessor {
   }
 
   @override
-  RealmList<ElementT> getList<ElementT>(RealmObject object, String propertyName) {
+  RealmList<ElementT> getList<ElementT>(RealmObjectBase object, String propertyName) {
     final propertyMeta = metadata[propertyName];
     final handle = realmCore.getListProperty(object, propertyMeta.key);
     final realm = object.realm;
-    return RealmListInternal.create(
+
+    // TODO: is there a better way?
```

Comment:
@nielsenko let's chat about this - this approach is pretty obnoxious but I couldn't find a way to bend the type system to do what I wanted it to.